### ISSUE TITLE
docker-compose: Server did not provide an image ID

### ISSUE THREAD

**Issue by hiaselhans** [containers/podman#11326]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I try to use podman with docker-compose and reuse my old config.

docker-compose with podman works for repository-images but seems to fail with locally-built images.

**Steps to reproduce the issue:**

1.

enable podman socket at /run/docker.sock

2.

*docker-compose.yml*
``` yml
version: '3.7'
services:
  gitea:
    build: "."
    environment:
      - DB_TYPE=postgres
      - DB_HOST=db:5432
      - DB_NAME=gitea
      - DB_USER=gitea
      - DB_PASSWD=gitea
    restart: always
    volumes:
      - git_data:/data
    ports:
      - 3000:3000
  db:
    image: postgres:alpine
    environment:
      - POSTGRES_USER=gitea
      - POSTGRES_PASSWORD=gitea
      - POSTGRES_DB=gitea
    restart: always
    volumes:
      - db_data:/var/lib/postgresql/data
    expose:
      - 5432
volumes:
  db_data:
  git_data:
```

*Dockerfile*
```
from gitea/gitea:latest

RUN echo "just to build something"
```

`docker-compose build`

**Describe the results you received:**

```
$ docker-compose build
db uses an image, skipping
Building gitea
Sending build context to Docker daemon  3.584kB
STEP 1/2: FROM gitea/gitea:latest
STEP 2/2: RUN echo "just to build something"
just to build something
COMMIT docker_gitea
Getting image source signatures
Copying blob sha256:b2d5eeeaba3a22b9b8aa97261957974a6bd65274ebd43e1d81d0a7b8b752b116
Copying blob sha256:940aa658ccb5e41d1d0c9694eeb3a4a9a281ac7ce4807a00b123988b0200b4ac
Copying blob sha256:18654675f84f98fc2059ed5abcf44a2f5140ea4de7d6765776a0de37b28be8c2
Copying blob sha256:e2351d9472e9d06e724e6e279fe6c3ff1607d323f09f44c8f184d2831394a1ad
Copying blob sha256:28822be09a0f444d51fc57ed2b4acbc4a63f1ec7798d1adf08b328bf7f1bbde6
Copying blob sha256:0869d4c93150748795550b85a3fba726178cb09a074088fdcf38fda3e51bfd51
Copying blob sha256:97d1c505a1112cf60d4502760043ce0c3762c194716e755f042651d6d90b37ad
Copying blob sha256:13d2687b7946da833edda21b8f7d3968a6579dc103ba80aeff49af34683be162
Copying config sha256:9cf6441c497d0a7fd9d6fe2c21a61f967e08271fcd2c942fe87ab62d4233c355
Writing manifest to image destination
Storing signatures
--> 9cf6441c497
Successfully tagged localhost/docker_gitea:latest
9cf6441c497d0a7fd9d6fe2c21a61f967e08271fcd2c942fe87ab62d4233c355
Successfully built 9cf6441c497d
Successfully tagged docker_gitea
Server did not provide an image ID. Cannot write /tmp/tmp1p546fld
ERROR: Service 'gitea' failed to build : Build failed
```

**Describe the results you expected:**

api-compatibility with docker

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:      3.3.0
API Version:  3.3.0
Go Version:   go1.17
Git Commit:   98f252a3a1a8f1ee00f9f96c6ba00500954b5093-dirty
Built:        Sat Aug 21 13:48:18 2021
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```host:
  arch: amd64
  buildahVersion: 1.22.3
  cgroupControllers: []
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: /usr/bin/conmon ist in conmon 1:2.0.29-1 enthalten
    path: /usr/bin/conmon
    version: 'conmon version 2.0.29, commit: 7e6de6678f6ed8a18661e1d5721b81ccee293b9b'
  cpus: 8
  distribution:
    distribution: arch
    version: unknown
  eventLogger: journald
  hostname: laptop
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.13.12-arch1-1
  linkmode: dynamic
  memFree: 480317440
  memTotal: 33501949952
  ociRuntime:
    name: crun
    package: /usr/bin/crun ist in crun 0.21-1 enthalten
    path: /usr/bin/crun
    version: |-
      crun version 0.21
      commit: c4c3cdf2ce408ed44a9e027c618473e6485c635b
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /etc/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: /usr/bin/slirp4netns ist in slirp4netns 1.1.12-1 enthalten
    version: |-
      slirp4netns version 1.1.12
      commit: 7a104a101aa3278a2152351a082a6df71f57c9a3
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.1
  swapFree: 8376545280
  swapTotal: 8393846784
  uptime: 21h 3m 41.74s (Approximately 0.88 days)
registries:
  search:
  - docker.io
store:
  configFile: /home/simon/.config/containers/storage.conf
  containerStore:
    number: 10
    paused: 0
    running: 0
    stopped: 10

```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```Name                     : podman
Version                  : 3.3.0-1
Beschreibung             : Tool and library for running OCI-based containers in pods
Architektur              : x86_64
URL                      : https://github.com/containers/podman
Lizenzen                 : Apache
Gruppen                  : Nichts
Stellt bereit            : Nichts
Hängt ab von             : cni-plugins  conmon  containers-common  device-mapper  iptables  libseccomp  crun  slirp4netns  libsystemd  fuse-overlayfs  libgpgme.so=11-64
Optionale Abhängigkeiten : podman-docker: for Docker-compatible CLI
                           btrfs-progs: support btrfs backend devices [Installiert]
                           catatonit: --init flag support
                           crun: support for unified cgroupsv2 [Installiert]
Benötigt von             : cockpit-podman
Optional für             : Nichts
In Konflikt mit          : Nichts
Ersetzt                  : Nichts
Installationsgröße       : 71,69 MiB
Packer                   : Morten Linderud <foxboron@archlinux.org>
Erstellt am              : Sa 21 Aug 2021 13:48:18 CEST
Installiert am           : Mi 25 Aug 2021 11:56:21 CEST
Installationsgrund       : Ausdrücklich installiert
Installations-Skript     : Nein
Verifiziert durch        : Signatur

```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

Yes

**Comment by hiaselhans** [comment]
i can confirm that images are built with different prefix:

### docker:

```
[simon@simonlaptop docker]$ COMPOSE_DOCKER_CLI_BUILD=0 && docker-compose build
db uses an image, skipping
Building gitea
Sending build context to Docker daemon  3.584kB
Step 1/2 : from gitea/gitea:latest
 ---> 7737d8e9cae8
Step 2/2 : RUN echo "just to build something"
 ---> Using cache
 ---> 3ce0117563a0
Successfully built 3ce0117563a0
Successfully tagged docker_gitea:latest
```

```
[simon@simonlaptop docker]$ docker image ls
REPOSITORY                                      TAG             IMAGE ID       CREATED          SIZE
docker_gitea                                    latest          3ce0117563a0   16 seconds ago   148MB
postgres                                        alpine          b8c450ae0903   3 days ago       192MB
gitea/gitea                                     latest          7737d8e9cae8   9 days ago       148MB
```

### podman:

```
[simon@simonlaptop docker]$ COMPOSE_DOCKER_CLI_BUILD=0 && docker-compose build
db uses an image, skipping
Building gitea
Sending build context to Docker daemon  3.584kB
STEP 1/2: FROM gitea/gitea:latest
STEP 2/2: RUN echo "just to build something"
just to build something
COMMIT docker_gitea
Getting image source signatures
Copying blob sha256:b2d5eeeaba3a22b9b8aa97261957974a6bd65274ebd43e1d81d0a7b8b752b116
Copying blob sha256:940aa658ccb5e41d1d0c9694eeb3a4a9a281ac7ce4807a00b123988b0200b4ac
Copying blob sha256:18654675f84f98fc2059ed5abcf44a2f5140ea4de7d6765776a0de37b28be8c2
Copying blob sha256:e2351d9472e9d06e724e6e279fe6c3ff1607d323f09f44c8f184d2831394a1ad
Copying blob sha256:28822be09a0f444d51fc57ed2b4acbc4a63f1ec7798d1adf08b328bf7f1bbde6
Copying blob sha256:0869d4c93150748795550b85a3fba726178cb09a074088fdcf38fda3e51bfd51
Copying blob sha256:97d1c505a1112cf60d4502760043ce0c3762c194716e755f042651d6d90b37ad
Copying blob sha256:cc10bb93deb416811ea5e65a827dcc3e2f83c307d422053a62ed70fc98c1f4d6
Copying config sha256:88236ef19ec297108ff9d329a5d04f86a8661ab51152313d2e5a21b85122efda
Writing manifest to image destination
Storing signatures
--> 88236ef19ec
Successfully tagged localhost/docker_gitea:latest
88236ef19ec297108ff9d329a5d04f86a8661ab51152313d2e5a21b85122efda
Successfully built 88236ef19ec2
Successfully tagged docker_gitea
Server did not provide an image ID. Cannot write /tmp/tmps0rm328_
ERROR: Service 'gitea' failed to build : Build failed
```

```
[simon@simonlaptop docker]$ docker image ls
REPOSITORY                                      TAG             IMAGE ID       CREATED         SIZE
gitea/gitea                                     latest          7737d8e9cae8   9 days ago      149MB
postgres                                        alpine          16a925851698   2 weeks ago     195MB
localhost/docker_gitea                          latest          9d33c7dc9282   5 days ago      149MB
```

however, starting the thing with `docker-compose up` works with podman after the build.

**Comment by rhatdan** [comment]
I believe this issue is fixed, reopen if I am mistaken.

**Comment by almereyda** [comment]
This persists on Fedora 35 when using the Podman user socket through `DOCKER_HOST` and Docker Compose with i.e. https://gitlab.com/omd_public/assemblage.

```
$ podman version
Version:      3.4.7
API Version:  3.4.7
Go Version:   go1.16.15
Built:        Mon Jul 11 00:35:25 2022
OS/Arch:      linux/amd64
```

I cannot reopen this issue. Should I recreate it?

**Comment by mheon** [comment]
Please try a more recent Podman. We just released 4.2 last week.
